File: ATM_Python_Sockets_MySQL/ServerTCP.py
import threading
import socket
import sys
import ServerThread

def run():
	PORT = 1230
	try:
		connectionSocket = socket.socket()
		connectionSocket.bind(('', PORT))
		connectionSocket.listen(5)
		print("Server is listening to port " + str(PORT))
	except socket.error as err:
		print("Connection problem" + str(err))
		sys.exit(1)

	while True:
		dataSocket, address = connectionSocket.accept()
		print("Received request from " + str(dataSocket.getsockname()))
		serverthread = threading.Thread(target=ServerThread.run, args=(dataSocket,))
		serverthread.start()


if __name__ == "__main__":
	thread = threading.Thread(target=run)
	thread.start()
	thread.join()

File: ATM_Java_Sockets_MySQL/ServerTCP.java
import java.io.IOException;
import java.net.ServerSocket;
import java.net.Socket;

public class ServerTCP extends Thread {
    private static final int PORT = 1230;
    
    public static void main(String[] args){
        ServerTCP server = new ServerTCP();
        server.start();
    }

    public void run(){
        while(true){
            try{
                ServerSocket connectionSocket = new ServerSocket(PORT);
                System.out.println("Server is listening to port: " + PORT);

                Socket dataSocket = connectionSocket.accept();
                System.out.println("Received request from " + dataSocket.getInetAddress());

                ServerThread thread = new ServerThread(dataSocket);
                thread.start();
                connectionSocket.close();
            } catch(IOException e){
				System.out.println(e.getMessage());
			}
        }
    }
}
